lib/Models/Terria.ts

```diff
@@ -857,18 +855,18 @@ export default class Terria {
           .map(item => <TimeVarying>item);
 
         // Load the items on the workbench
-        for (let model of newItems) {
-          if (ReferenceMixin.is(model)) {
-            promises.push(model.loadReference());
-            model = model.target || model;
-          }
-
-          if (Mappable.is(model)) {
-            promises.push(model.loadMapItems());
-          }
-        }
+        return Promise.all(
+          newItems.map(async model => {
+            if (ReferenceMixin.is(model)) {
+              await model.loadReference();
+              model = model.target || model;
+            }
 
-        return Promise.all(promises).then(() => undefined);
+            if (Mappable.is(model)) {
+              await model.loadMapItems();
```

Comment:
This could be quite slow. What's the problem with loading multiple workbench items at once?

Reply:
Ohhhh. Whoops. Just realised what this is doing. It synchronises loadReference -> loadMap for each, but runs every item in parallel to every other item. That's exactly what it should do.